Neo4j.Driver/Neo4j.Driver/Internal/Auth/AuthToken.cs

```diff
@@ -31,23 +32,62 @@ internal class AuthToken : IAuthToken
 
     public AuthToken(IDictionary<string, object> content)
     {
-        Content = content ?? throw new ArgumentNullException(nameof(content));
+        content = content ?? throw new ArgumentNullException(nameof(content));
+        Content = new Dictionary<string, object>();
+        foreach (var (key, value) in content)
+        {
+            if (value is not null)
+            {
+                Content[key] = value;
+            }
+        }
     }
 
     public IDictionary<string, object> Content { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is AuthToken a && Equals(a);
+    }
+
+    protected bool Equals(AuthToken other)
+    {
+        if (ReferenceEquals(this, other))
+        {
+            return true;
+        }
+
+        var equal =
+            Content.Count == other.Content.Count &&
+            Content.All(kvp => other.Content[kvp.Key].Equals(kvp.Value));
+
+        return equal;
+    }
+
+    public override int GetHashCode()
+    {
+        return Content != null ? Content.GetHashCode() : 0;
+    }
+
+    public override string ToString()
+    {
+        return string.Join("; ", Content.Select(kvp => $"{kvp.Key}: {kvp.Value}"));
```

Comment:
I don't think we should stringify the contents of auth tokens. maybe the the type but that's all

Reply:
Removed - it was only for debugging anyway.